release_notes.md

```diff
@@ -1,3 +1,54 @@
+***********************************
+# Release Notes for OpenMDAO 3.12.0
+
+September 15, 2021
+
+OpenMDAO 3.12.0 provides a transitional release as we change the way the specification of indices are handled.
+In the future, specification of indices will be similar to the way it is done in Numpy, as opposed to an OpenMDAO-specific method.
+See [POEM 053](https://github.com/OpenMDAO/POEMs/blob/master/POEM_053.md) and [POEM 054](https://github.com/OpenMDAO/POEMs/blob/master/POEM_054.md)
+for more information.
+
+## New Deprecations
+
+- Behavior of `'rel'` step_calc for finite difference is changed and issues a deprecation warning if used, see [#2209](https://github.com/OpenMDAO/OpenMDAO/pull/2209) for more information.
+
+## Backwards Incompatible API Changes
+
+- Disabled the option to use coloring on approximated semi-total derivatives. [#2245](https://github.com/OpenMDAO/OpenMDAO/pull/2245)
+
+## Backwards Incompatible Non-API Changes
+
+## New Features
+
+- Improved behavior of relative step_calc for finite difference in accordance with [POEM 051](https://github.com/OpenMDAO/POEMs/blob/master/POEM_051.md). [#2209](https://github.com/OpenMDAO/OpenMDAO/pull/2209)
+- Add a check to openmdao check to report any options in the model that are not serializable. [#2225](https://github.com/OpenMDAO/OpenMDAO/pull/2225)
+- Shift-click can now close all persistent Node Info windows in the N2 viewer. [#2226](https://github.com/OpenMDAO/OpenMDAO/pull/2226)
+- Update to src_indices indexing to behave like numpy indexing (old indexing is still there but deprecated) [#2235](https://github.com/OpenMDAO/OpenMDAO/pull/2235)
+- Added code to issue warnings in cases where apparently flat indices index into non-flat source arrays when the flat_src_indices or flat_indices flags have not been set to True. [#2248](https://github.com/OpenMDAO/OpenMDAO/pull/2248)
+- Added a converter from old OpenMDAO indexing format to the new numpy-like one. [#2267](https://github.com/OpenMDAO/OpenMDAO/pull/2267)
+
+## Bug Fixes
+
+- Fixed the hang described in [#2191](https://github.com/OpenMDAO/OpenMDAO/pull/2191) so all procs will call multi_proc_exception_check()  [#2208](https://github.com/OpenMDAO/OpenMDAO/pull/2208)
+- Added support for multidimensional shapes in EQConstraintComp and BalanceComp. [#2210](https://github.com/OpenMDAO/OpenMDAO/pull/2210)
+- Fixed bug where an exception was raised if you tried to get a value for an input that was connected to a source with a src_indices slice. [#2217](https://github.com/OpenMDAO/OpenMDAO/pull/2217)
+- Fixed a bug in the interpolation bracketing algorithm that caused a hysteresis in the analytic derivative when interpolating on one of the table's gridpoints. [#2230](https://github.com/OpenMDAO/OpenMDAO/pull/2230)
+- Fixed issue with directional derivs and distributed vars [#2233](https://github.com/OpenMDAO/OpenMDAO/pull/2233)
```

Comment:
"Fixed issue with directional derivs and distributed vars " -> "Fixed a bug that caused an IndexError to be raised when using a directional derivative during check_partials on an MPI model."